### DISCUSSION TITLE
Hide/Fold Frontmatter?

### DISCUSSION THREAD

**Discussion by oschrenk** [epwalsh/obsidian.nvim#591] [General]
Hi,  

I usually have a few lines of frontmatter properties (especially for my daily notes). Most of them are automatically created

Is there a way to hide them or fold them away when I open the note?
Do you use any plugins for that?

Ideally everything before the first `#` is folded away?

**Comment by stefangstark** [comment]
I've been searching for something similar and hacked an autocmd instead. When entering a markdown file it detects if frontmatter exists and puts the text after at the top of the file

```
vim.api.nvim_create_autocmd({"BufEnter", "BufWinEnter"}, {
  pattern = "*.md",
  callback = function()
    local search = vim.fn.search('\\%^---', 'n')
    if search == 0 then
      return
    end
    vim.cmd("2/---/2")
    local old_option = vim.opt.scrolloff
    vim.opt.scrolloff = 0
    vim.cmd("normal! zt")
    vim.opt.scrolloff = old_option
  end
})
```

Not a perfect solution but at least I don't need to look at frontmatter when opening a new file

**Reply by oschrenk** [reply]
Nice!

I ended up using

```
keys = {
    {
      "<leader>ot",
      mode = { "n", "x", "o" },
      function()
        local client = require("obsidian").get_client()
        local note = client:daily(0)
        client:open_note(note, { sync = true })
        vim.api.nvim_win_set_cursor(0, { 11, 0 })
        local key = vim.api.nvim_replace_termcodes("z<cr>", true, false, true)
        vim.api.nvim_feedkeys(key, "n", true)
      end,
      desc = "Today",
    },
   ...
```

But I like your approach better to find `---` and will probably mix both solutions moving forward. 

Thank you!
